File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/application/service/auth/TestEnvironmentUserAuthenticator.java
package kookmin.software.capstone2023.timebank.application.service.auth;

import kookmin.software.capstone2023.timebank.domain.model.AccountType;

public class TestEnvironmentUserAuthenticator implements UserAuthenticator {
    private final long userId;
    private final long accountId;
    private final AccountType accountType;

    public TestEnvironmentUserAuthenticator(long userId, long accountId, AccountType accountType) {
        this.userId = userId;
        this.accountId = accountId;
        this.accountType = accountType;
    }

    @Override
    public UserAuthenticator.AuthenticationData authenticate(String accessToken) {
        return new UserAuthenticator.AuthenticationData(userId, accountId, accountType);
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/application/service/auth/TestEnvironmentUserAuthenticator.kt
package kookmin.software.capstone2023.timebank.application.service.auth

import kookmin.software.capstone2023.timebank.domain.model.AccountType

class TestEnvironmentUserAuthenticator(
    private val userId: Long,
    private val accountId: Long,
    private val accountType: AccountType,
) : UserAuthenticator {
    override fun authenticate(accessToken: String): UserAuthenticator.AuthenticationData {
        return UserAuthenticator.AuthenticationData(
            userId = userId,
            accountId = accountId,
            accountType = accountType,
        )
    }
}
